ClassCastException with adoptopenjdk/openjdk11-openj9

Hi

we upgrade to the current 1.4 version and got this error upon startup:

```
textPayload: "java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at co.elastic.apm.agent.bci.AgentMain.init(AgentMain.java:78)
	at co.elastic.apm.agent.bci.AgentMain.premain(AgentMain.java:48)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:513)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:525)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/java.lang.J9VMInternals.ensureError(J9VMInternals.java:191)
	at java.base/java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:180)
	at java.base/java.lang.Class.forNameImpl(Native Method)
	at java.base/java.lang.Class.forName(Class.java:340)
	at co.elastic.apm.agent.metrics.builtin.JvmGcMetrics.bindTo(JvmGcMetrics.java:66)
	at co.elastic.apm.agent.metrics.builtin.JvmGcMetrics.start(JvmGcMetrics.java:41)
	at co.elastic.apm.agent.impl.ElasticApmTracer.<init>(ElasticApmTracer.java:157)
	at co.elastic.apm.agent.impl.ElasticApmTracerBuilder.build(ElasticApmTracerBuilder.java:108)
	at co.elastic.apm.agent.bci.ElasticApmAgent.initialize(ElasticApmAgent.java:90)
	... 12 more
Caused by: java.lang.ClassCastException: com.ibm.lang.management.internal.ExtendedThreadMXBeanImpl incompatible with com.sun.management.ThreadMXBean
	at co.elastic.apm.agent.metrics.builtin.JvmGcMetrics$HotspotAllocationSupplier.<init>(JvmGcMetrics.java:75)
	at co.elastic.apm.agent.metrics.builtin.JvmGcMetrics$HotspotAllocationSupplier.<clinit>(JvmGcMetrics.java:73)
```

For our project with have `adoptopenjdk/openjdk11-openj9` as base image. It seems that the given metric goes to much into VM specific details or should back-off if it does not find it.

--- a/apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/builtin/JvmGcMetrics.java
+++ b/apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/builtin/JvmGcMetrics.java
@@ -60,10 +60,12 @@ public class JvmGcMetrics implements LifecycleListener {
 
         try {
             // only refer to hotspot specific class via reflection to avoid linkage errors
-            Class.forName("com.sun.management.ThreadMXBean");
-            // in reference to JMH's GC profiler (gc.alloc.rate)
-            registry.add("jvm.gc.alloc", Collections.<String, String>emptyMap(),
-                (DoubleSupplier) Class.forName(getClass().getName() + "$HotspotAllocationSupplier").getEnumConstants()[0]);
+            final Class<?> sunBeanClass = Class.forName("com.sun.management.ThreadMXBean");
+            if (sunBeanClass.isInstance(ManagementFactory.getThreadMXBean())) {
+                // in reference to JMH's GC profiler (gc.alloc.rate)
+                registry.add("jvm.gc.alloc", Collections.<String, String>emptyMap(),
+                    (DoubleSupplier) Class.forName(getClass().getName() + "$HotspotAllocationSupplier").getEnumConstants()[0]);
+            }
         } catch (ClassNotFoundException ignore) {
         }
     }

--- a/apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/builtin/JvmGcMetrics.java
+++ b/apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/builtin/JvmGcMetrics.java
@@ -61,6 +61,8 @@ public class JvmGcMetrics implements LifecycleListener {
         try {
             // only refer to hotspot specific class via reflection to avoid linkage errors
             final Class<?> sunBeanClass = Class.forName("com.sun.management.ThreadMXBean");
+            // J9 does contain com.sun.management.ThreadMXBean in classpath
+            // but the actual MBean it uses (com.ibm.lang.management.internal.ExtendedThreadMXBeanImpl) does not implement it
             if (sunBeanClass.isInstance(ManagementFactory.getThreadMXBean())) {
                 // in reference to JMH's GC profiler (gc.alloc.rate)
                 registry.add("jvm.gc.alloc", Collections.<String, String>emptyMap(),

